### PR TITLE
Add PageLogFormat, MaxLogSize, JobPrivateAccess and JobPrivateValues

### PR DESCRIPTION
**Pull request by timbirk**
Hi,

First, thanks for a nice, readable module with good testing!

This PR adds a few options that I find useful: PageLogFormat, MaxLogSize, JobPrivateAccess and JobPrivateValues and related docs / tests. By default the configuration values are as they originally were. 

In future I might open further PRs as and when the need arises to set other values. Any feedback or criticism is always appreciated.

Thanks,

Tim Birkett.

Legal statement:

By committing to this project I transfer the full copyright for my contributions
to the current project maintainer as per the project's LICENSE file.

### CODE REVIEW

**Comment by timbirk** [comment]
@leoarnold Just seen another PR request with your comment:
> We have a larger feature in preparation which will superseed your contribution and contain all the tedious extra mileage of testing and documentation.

I'm guessing this may be covered in the new feature?

**Comment by leoarnold** [comment]
Dear @timbirk,

first of all thank you for your pull request. It is rare to see such a well prepared one with specs and docs.

My comment from another PR relates to all parameters enclosed in pseudo-XML tags like `<Location />` or `<Policy default>`. Once we open that can of worms, people will request to be able to configure all kinds of parameters in there (and rightfully so). So what you suggest as `$job_private_access` may be `$policy['default]['job_private_access']` from someone else's perspective. The question I'm currently meditating on is how to model the attributes of `Class['cups']` in a one-size-fits-all way, and what would be the desirable defaults there.

Also, development was kind of stalled for the past months because Puppet's new i18n feature had a weird concurrency bug which failed the unit tests, but it seems fixed in Puppet 5.5+.

Long story short, your PR will definitely be merged in one way or another and I'll make sure you'll be credited on the GitHub contributors page of this repo, but it might take a while.

Also, thanks for letting me know that this module is used at ITV. I think ITV the first known user outside the educational system.

**Comment by timbirk** [comment]
Thanks for the kind words @leoarnold,

I have the fun of working in the Back Office and Finance division at ITV (Exciting... I know) and this module made it really easy to get CUPS configured in AWS to introduce some reliability when printing to office based secure printers from Java applications.

I get what you mean, allowing customisation of every available option is a big can of worms (especially when it comes to CUPS). From what I can gather, the config format is very very similar to Apache (httpd) configuration.

I'd be happy to expand this PR to improve configurability with sensible defaults (primarily CUPS 
 / existing module defaults) and possibly abstract things like Policies and Locations out to defined resources (if it makes sense to) which could be created from hiera? Starting with `man cupsd.conf` might be the wisest first step.

I might also meditate on it and start some discussion ideas in the near(ish) future if you're comfortable with that?
